cswinrt/code_writers.h

```diff
@@ -5080,7 +5080,8 @@ public static unsafe void DisposeAbiArray(object box) => MarshalInspectable<obje
                             {
                                 w.write(
                                     ".As<IUnknownVftbl>(GuidGenerator.GetIID(typeof(%).GetHelperType()))",
-                                    bind<write_type_name>(get_type_semantics(get_default_interface(type)), typedef_name_type::Projected, false));
+                                    bind<write_type_name>(get_type_semantics(get_default_interface(type)), 
+                                        settings.component ? typedef_name_type::ABI : typedef_name_type::Projected, false));
```

Comment:
I feel like you want typedef_name_type::ccw here which should handle component and non component.